static/js/compose.js

```diff
@@ -987,7 +987,7 @@ exports.initialize = function () {
         if (email === undefined) {
             return;
         }
-
+        const user_id = people.get_user_id(email);
```

Comment:
We can do better than this -- let's change that `useremail` property to be a `data-user-id` field in the DOM instead.  That's our consistent pattern for referring to users everywhere, and seems like we should do it here too.  And then we don't need some of these checks at all.

Reply:
@timabbott I had a doubt here, why `get_active_user_for_email` is used here as we do not call this codepath if we try to mention a deactivated user by writing the syntax by ourselves (as deactivated users are not present in typeaheads and can't be mentioned actually). I guess that this was there to handle the case where deactivated user has same email as that of a active user, but I am not sure about this.

I am asking this because, I wanted to know whether we need to write a separate function like
`get_active_user_by_id` because if it was to handle the duplicate email case, then I assume we do not need to use separate function for active because I think two ids cannot be same even when one is for deactivated and other is for activated users.